src/connection-manager/connections.ts

```diff
@@ -97,7 +99,28 @@ async function setHandlers(
         )
       }
     })
-    ws.on('close', () => {
+    ws.on('close', async (code, reason) => {
+      if (connectionsInitialized) {
+        log.error(
+          `Websocket closed for ${ws.url} on ${
+            networks ?? 'unknown network'
+          } with code ${code} and reason ${reason.toString('utf-8')}.`,
+        )
+        if (CLOSING_CODES.includes(code)) {
+          log.info(
+            `Reconnecting to ${ws.url} on ${networks ?? 'unknown network'}...`,
+          )
+          // Open a new Websocket connection for the same url
+          const newWS = new WebSocket(ws.url, { handshakeTimeout: WS_TIMEOUT })
+          // Clean up the old Websocket connection
+          connections.delete(ip)
+          ws.terminate()
+          resolve()
+
+          await setHandlers(ip, newWS, networks, isInitialNode)
+          return
```

Comment:
Can you add a comment explaining why you return here instead of allowing the rest of the logic to run?

Reply:
The return statement comes inside a nested if statement so I think it suffice to just add a comment.